### PR TITLE
Use Cases: Add contribute to ripple-lib

### PR DESCRIPTION
**Pull request by jhaaaa**
Add a use case for how to contribute to `ripple-lib`

### CODE REVIEW

File: content/use-cases/contribute-code-to-ripple-lib.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Set up and run a `rippled` server](manage-the-rippled-server.html)
+
+RippleAPI is an API for interacting with the XRP Ledger. The core peer-to-peer server that manages the XRP Ledger is `rippled`. Set up and run a `rippled` server to understand its developer experience and functionality. Anyone can run their own `rippled` server that follows the network and keeps a complete copy of the XRP Ledger.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Get a sandbox XRP Ledger account](xrp-test-net-faucet.html)
```
Comment (intelliot) [inline content/use-cases/contribute-code-to-ripple-lib.md]:
```suggestion
## [Get a Test Net XRP Ledger account](xrp-test-net-faucet.html)
```

File: content/use-cases/contribute-code-to-ripple-lib.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Use the XRP Ledger Test Net to get a sandbox account. Connect your `rippled` server to the Test Net to make test calls and get to know the XRP Ledger.
```
Comment (intelliot) [inline content/use-cases/contribute-code-to-ripple-lib.md]: I usually call it a test account, Test Net account, or Test Net address. I haven't seen "sandbox" used.

**Review by intelliot — COMMENTED** [review]
This is a great start!

I would add that there are a number of issues tagged with "help wanted" - this would be a good place for someone to start contributing: https://github.com/ripple/ripple-lib/issues?utf8=%E2%9C%93&q=label%3A%22help+wanted%22

File: content/use-cases/contribute-code-to-ripple-lib.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## <a href="https://github.com/ripple/ripple-lib/pulls" target="_blank">Contribute code <i class="fa fa-external-link" aria-hidden="true"></i></a><!--#{ fix for md highlighting_ #}-->
+
+Now that you have a handle on `ripple-lib`, you may have ideas for how to improve it.
+
+Perhaps you’re developing on the XRP Ledger and want to contribute some code that enables `ripple-lib` to provide a feature your application needs.
+
+Need some inspiration? Take a look at our list of [Help Wanted issues](https://github.com/ripple/ripple-lib/issues?utf8=%E2%9C%93&q=label%3A%22help+wanted%22).
```
Comment (jhaaaa) [inline content/use-cases/contribute-code-to-ripple-lib.md]: Fantastic idea to add this, @intelliot!

File: content/use-cases/contribute-code-to-ripple-lib.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+## [Run your first `ripple-lib` script](get-started-with-rippleapi-for-javascript.html#first-rippleapi-script)
+
+Examine and run the `get-account-info.js` script. Use it to get a feel for how RippleAPI scripts work and to verify that your RippleAPI interface is working.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## Familiarize yourself with `ripple-lib`'s coding style<!--#{ link up once we have some a ripple-lib coding style page #}-->
+
+Before you start contributing code to `ripple-lib,` take some time to familiarize yourself with the coding standards used in the `ripple-lib` repo. These standards gradually evolve and propagate through code reviews. Some aspects are enforced more strictly than others. ***TODO: Here are the coding standards for rippled - specific to C++: https://github.com/ripple/rippled/blob/develop/docs/CodingStyle.md. Do we want to modify for JavaScript and ripple-lib and provide it in the ripple-lib repo?***
```
Comment (jhaaaa) [inline content/use-cases/contribute-code-to-ripple-lib.md]: @intelliot - do we want to provide ripple-lib-specific coding style info before we publish this use case? Or should we publish the use case without this step -- and update later? Any thoughts?

Comment (intelliot) [reply content/use-cases/contribute-code-to-ripple-lib.md]: We have a linter (`tslint`) that catches most inconsistencies automatically. The rest get caught in code review. I'm fine with leaving out mention of a coding style for now; it hasn't been a problem yet.

File: content/use-cases/contribute-code-to-ripple-lib.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# Contribute Code to ripple-lib
+
+Want to contribute code or a bug report to help improve `ripple-lib`, the official client library for [RippleAPI](rippleapi-reference.html)? RippleAPI is a JavaScript API for interacting with the XRP Ledger. Here’s a roadmap to the high-level tasks that’ll have you reviewing code and functionality in no time. ***TODO: What do you think of the use of ripple-lib vs RippleAPI throughout this use case? Are these being used correctly?***
```
Comment (intelliot) [inline content/use-cases/contribute-code-to-ripple-lib.md]: In some contexts they can be used interchangeably. I think of `ripple-lib` as being more specific - [this JS library](https://github.com/ripple/ripple-lib) - while `RippleAPI` includes `ripple-lib` and encompasses other APIs that are being developed and will be released in the near future (~2-3 months).

File: content/use-cases/contribute-code-to-ripple-lib.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## <a href="https://github.com/ripple/ripple-lib" target="_blank">Access the `ripple-lib` repo <i class="fa fa-external-link" aria-hidden="true"></i></a> <!--#{ fix for md highlighting_ #}-->
+
+`ripple-lib` is an open-source project. You can take a look at `ripple-lib` code simply by accessing the `ripple-lib` GitHub repo. Before contributing or reporting bugs, we recommend that you get to know the code and developer experience by performing the following tasks.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Set up and run a `rippled` server](manage-the-rippled-server.html)
```
Comment (intelliot) [inline content/use-cases/contribute-code-to-ripple-lib.md:17]: This isn't necessary for using ripple-lib (as it has some offline functionality and can connect to public rippled servers), but it is nice.

File: content/use-cases/contribute-code-to-ripple-lib.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+RippleAPI is an API for interacting with the XRP Ledger. The core peer-to-peer server that manages the XRP Ledger is `rippled`. Set up and run a `rippled` server to understand its developer experience and functionality. Anyone can run their own `rippled` server that follows the network and keeps a complete copy of the XRP Ledger.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Get a Test Net XRP Ledger account](xrp-test-net-faucet.html)
+
+Use the XRP Ledger Test Net to get a test account. Connect your `rippled` server to the Test Net to make test calls and get to know the XRP Ledger.
```
Comment (intelliot) [inline content/use-cases/contribute-code-to-ripple-lib.md]:
```suggestion
Use the XRP Test Net Faucet to get a test account on the test network. Connect your `rippled` server to the Test Net to make test calls and get to know the XRP Ledger.
```

The faucet makes it easy to get a test account on the test network.

File: content/use-cases/contribute-code-to-ripple-lib.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Set up your `ripple-lib` development environment](get-started-with-rippleapi-for-javascript.html#environment-setup)
+
+`ripple-lib` requires Node.js and a few dependencies that you can get using npm (Node Package Manager). Also, be sure to create your own fork of the `ripple-lib` repository on GitHub so you can contribute pull requests to the official repo.
```
Comment (intelliot) [inline content/use-cases/contribute-code-to-ripple-lib.md]: I recommend using [Node.js v10 LTS](https://nodejs.org/) and [Yarn](https://yarnpkg.com/).

**Comment by jhaaaa** [comment]
Updated content based on Elliot's feedback. 👍 

@intelliot, if it looks good, may I have your approval?

@mDuo13, please feel free to provide any feedback/guidance if something jumps out at you. Thank you!

**Review by intelliot — APPROVED** [review]
Thanks for writing this!

File: dactyl-config.yml [hunk 9]
```diff
@@ -2507,6 +2507,15 @@ pages:
         targets:
             - local
 
+    -   md: use-cases/contribute-code-to-ripple-lib.md
+        html: contribute-code-to-ripple-lib.html
+        funnel: Use Cases
+        template: template-use-case.html
+        useful_background:
+            - consensus.html
+            - amendments.html #TODO add to useful_background: stand-alone mode, rippled.cfg reference (future feature)
```
Comment (mDuo13) [inline dactyl-config.yml]: suggested useful background:
- rippleapi-reference.html
- get-started-with-rippleapi-for-javascript.html

**Review by mDuo13 — APPROVED** [review]
I left a couple suggestions to add to the "Useful Background" area for this new use case. Otherwise, looks fantastic!

File: dactyl-config.yml (hunk 9 above)
Comment (jhaaaa) [reply dactyl-config.yml]: This is great - thank you for noticing this. I think I just copied the existing `useful_background` values from the `rippled` use case - no bueno. These are much more USEFUL. 🙌

**Comment by jhaaaa** [comment]
The build is failing due to a broken external link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode#Anchor_1.

I can access this link directly, as well as by clicking the link in my locally built docs.

**Pull request merged**